packages/rocketchat-livechat/imports/server/rest/upload.js

```diff
@@ -0,0 +1,76 @@
+import Busboy from 'busboy';
+import LivechatVisitors from '../../../server/models/LivechatVisitors';
+
+RocketChat.API.v1.addRoute('livechat/upload/:rid', {
```

Comment:
this endpoint has no checks regarding file types, size nor if it is allowed.

although the regular client flow is calling `livechat:validateFileUpload` method which does such validations before calling this endpoint, this endpoint cannot be unprotected as it currently is ..

so I suggest removing the `livechat:validateFileUpload` method and doing all his validations on this endpoint.

Reply:
Fixed.